src/dotnet/APIView/APIView/Model/TokenTreeModel.cs

```diff
@@ -0,0 +1,250 @@
+            set { RenderClasses = value ?? new HashSet<string>(); }
+        }
+        [JsonPropertyName("v")]
+        public string Value { get; set; } = string.Empty;
+        [JsonPropertyName("i")]
+        public string Id { get; set; }
+        [JsonPropertyName("k")]
+        public StructuredTokenKind Kind { get; set; }
+        [JsonIgnore]
+        public HashSet<string> Tags { get; set; } = new HashSet<string>();
+        [JsonIgnore]
+        public Dictionary<string, string> Properties { get; set; } = new Dictionary<string, string>();
+        [JsonIgnore]
+        public HashSet<string> RenderClasses { get; set; } = new HashSet<string>();
+
+        public StructuredToken()
+        {
+            new StructuredToken(string.Empty);
+        }
+
+        public StructuredToken(string value)
+        {
+            Value = value;
+            Kind = StructuredTokenKind.Content;
+        }
+
+        public static StructuredToken CreateLineBreakToken()
+        {
+            var token = new StructuredToken();
+            token.Kind = StructuredTokenKind.LineBreak;
+            return token;
+        }
+
+        public static StructuredToken CreateEmptyToken(string id = null)
+        {
+            var token = new StructuredToken();
+            token.Id = id;
+            token.Kind = StructuredTokenKind.Content;
+            return token;
+        }
+
+        public static StructuredToken CreateSpaceToken()
+        {
+            var token = new StructuredToken();
+            token.Kind = StructuredTokenKind.NoneBreakingSpace;
+            return token;
+        }
+
+        public static StructuredToken CreateParameterSeparatorToken()
+        {
+            var token = new StructuredToken();
+            token.Kind = StructuredTokenKind.ParameterSeparator;
+            return token;
+        }
+
+        public static StructuredToken CreateTextToken(string value, string id = null)
+        {
+            var token = new StructuredToken(value);
+            if (!string.IsNullOrEmpty(id))
+            {
+                token.Id = id;
+            }
+            token.RenderClasses.Add("text");
+            return token;
+        }
+
+        public static StructuredToken CreateKeywordToken(string value)
+        {
+            var token = new StructuredToken(value);
+            token.RenderClasses.Add("keyword");
+            return token;
+        }
+
+        public static StructuredToken CreateKeywordToken(SyntaxKind syntaxKind)
+        {
+            return CreateKeywordToken(SyntaxFacts.GetText(syntaxKind));
+        }
+
+        public static StructuredToken CreateKeywordToken(Accessibility accessibility)
+        {
+            return CreateKeywordToken(SyntaxFacts.GetText(accessibility));
+        }
+
+        public static StructuredToken CreatePunctuationToken(string value)
+        {
+            var token = new StructuredToken(value);
+            token.RenderClasses.Add("punc");
```

Comment:
Are there a set of well known values for RenderClasses? If so, should these be const values or an enum?

Reply:
The request here is to leave it up to the language classes. So they have freedom to change the rendering without bothering other languages or APIView main logic.